lib/src/chatbox.dart

```diff
@@ -52,6 +53,16 @@ class MessageActionEvent {
     message = Message.fromJson(json['message']);
 }
 
+enum UrlNavigationAction { allow, deny }
+
+class UrlNavigationRequest {
+  final String? url;
```

Comment:
What's the *practical* use of an UrlNavigationRequest with a null url?

Reply:
InAppWebView sometimes returns null for the url so it will not always be a String the result could be null.
I did a little more research on how InAppWebview works with urls and they do use a Uri object this means they parse the Url String to a Uri in their implementation.

This means that invalid urls won't work with there implementation, so creating a special url is only an option if it is also a valid Uri.

More information can be found here: https://inappwebview.dev/docs/webview/in-app-webview/#handle-custom-urls

I think this invalidates my previous comment about why it should be a String looking at this I do feel making it a Nullable Uri should be better. And custom urls can be handled like described in the link above. 

What do you think would be best?

Reply:
If you want we could also setup a meeting to discuss an maybe get this issue resolved a bit quicker?

Reply:
I believe that a non-nullable string is the best solution here, and here's why:
While not all Strings can be valid Uris, thus resulting in a null Uri, the same is not true for Uris, which are all valid Strings, so it's better to have a non-nullable String, and then let the user decide what to do with that string.

As I see from the documentation of the InAppWebView, if an Uri is invalid, the shouldOverrideUrlLoading method is not called anyway, so it's not a problem.
And the fact that the Uri can be null in the NavigationAction.URLRequest does not affect this functionality, as I'm sure that shouldOverrideUrlLoading will not be called with a null Uri (if you read the docs, a null Uri is equivalent to "about:blank").

Reply:
Good point, I've now changed it to a non nullable String